masonry_core/src/core/widget_pod.rs

```diff
@@ -13,89 +15,120 @@ pub struct WidgetPod<W: ?Sized> {
+    pub transform: Affine,
+    /// The disabled state the widget will be created with.
+    pub disabled: bool,
+}
+
 // TODO - This is a simple state machine that lets users create WidgetPods
 // without immediate access to the widget arena. It's very inefficient
 // and leads to ugly code. The alternative is to force users to create WidgetPods
 // through context methods where they already have access to the arena.
 // Implementing that requires solving non-trivial design questions.
 
-pub(crate) struct CreateWidget<W: ?Sized> {
-    pub(crate) widget: Box<W>,
-    pub(crate) options: WidgetOptions,
-    pub(crate) properties: Properties,
-}
-
 enum WidgetPodInner<W: ?Sized> {
-    Create(CreateWidget<W>),
+    Create(NewWidget<W>),
     Inserted,
 }
 
-impl<W: Widget> WidgetPod<W> {
-    /// Create a new widget pod.
-    ///
-    /// In a widget hierarchy, each widget is wrapped in a `WidgetPod`
-    /// so it can participate in layout and event flow. The process of
-    /// adding a child widget to a container should call this method.
+impl<W: Widget> NewWidget<W> {
+    /// Create a new widget.
     pub fn new(inner: W) -> Self {
         Self::new_with_id(inner, WidgetId::next())
     }
 
-    /// Create a new widget pod with fixed id.
+    /// Create a new widget with fixed id.
     pub fn new_with_id(inner: W, id: WidgetId) -> Self {
-        Self::new_with_id_and_options(Box::new(inner), id, WidgetOptions::default())
+        Self {
+            widget: Box::new(inner),
+            id,
+            options: WidgetOptions::default(),
+            properties: Properties::default(),
+        }
     }
 
-    /// Create a new widget pod with properties.
+    /// Create a new widget with properties.
     pub fn new_with_props(inner: W, props: Properties) -> Self {
-        Self::new_with(
-            Box::new(inner),
-            WidgetId::next(),
-            WidgetOptions::default(),
-            props,
-        )
+        Self {
+            widget: Box::new(inner),
+            id: WidgetId::next(),
+            options: WidgetOptions::default(),
+            properties: props,
+        }
     }
-}
 
-impl<W: Widget + ?Sized> WidgetPod<W> {
-    /// Create a new widget pod with custom options.
-    pub fn new_with_options(inner: Box<W>, options: WidgetOptions) -> Self {
-        Self::new_with_id_and_options(inner, WidgetId::next(), options)
+    /// Create a new widget with custom options.
+    pub fn new_with_options(inner: W, options: WidgetOptions) -> Self {
+        Self {
+            widget: Box::new(inner),
+            id: WidgetId::next(),
+            options,
+            properties: Properties::default(),
+        }
     }
 
-    /// Create a new widget pod with custom options and a pre-set [`WidgetId`].
-    pub fn new_with_id_and_options(inner: Box<W>, id: WidgetId, options: WidgetOptions) -> Self {
+    /// Create a new widget with custom options and custom [`Properties`].
```

Comment:
link options if you link properties.  Also, should call out the custom `WidgetId` as well?